zerver/views/auth.py

```diff
@@ -851,3 +872,24 @@ def password_reset(request: HttpRequest, **kwargs: Any) -> HttpResponse:
                                  template_name='zerver/reset.html',
                                  password_reset_form=ZulipPasswordResetForm,
                                  post_reset_redirect='/accounts/password/reset/done/')
+
+@csrf_exempt
+def saml_sp_metadata(request: HttpRequest, **kwargs: Any) -> HttpResponse:
+    """
+    This is the view function for generating our SP metadata
+    for SAML authentication.
+    Taken from https://python-social-auth.readthedocs.io/en/latest/backends/saml.html
+    """
+    if not saml_auth_enabled():
+        return HttpResponse(content='SAML not enabled.',
+                            content_type='text/plain')
```

Comment:
I think this should use `redirect_to_config_error`.

Reply:
Done, in a new commit.